internal/component/database_observability/mysql/component.go

```diff
@@ -369,12 +384,20 @@ func enableOrDisableCollectors(a Arguments) map[string]bool {
 }
 
 func (c *Component) startCollectors(serverUUID string, engineVersion string, parsedEngineVersion semver.Version) error {
+	// Best-effort start: try building/starting every enabled collector and aggregate errors.
+	var startErrors []string
+
+	logStartError := func(collectorName, action string, err error) {
+		wrapped := fmt.Errorf("failed to %s %s collector: %w", action, collectorName, err)
+		level.Error(c.opts.Logger).Log("msg", wrapped.Error())
+		startErrors = append(startErrors, wrapped.Error())
+	}
+
 	var cloudProviderInfo *database_observability.CloudProvider
 	if c.args.CloudProvider != nil && c.args.CloudProvider.AWS != nil {
 		arn, err := arn.Parse(c.args.CloudProvider.AWS.ARN)
 		if err != nil {
 			level.Error(c.opts.Logger).Log("msg", "failed to parse AWS cloud provider ARN", "err", err)
-			return err
```

Comment:
Hmm wondering if this could be added to the list of errors as well, although shouldn't prevent startup.